### ISSUE TITLE
Can't use package name "aspect.js" from Typescript project

### ISSUE THREAD

**Issue by matthewadams** [mgechev/aspect.js#68]
## TL;DR

`tsc` fails on `import`ing `aspect.js`, and @Microsoft won't fix it because it's working as intended.  Ergo, the project name `aspect.js` must be changed or its type mappings must be changed to be spec compliant.

## L;R
The package name `aspect.js` is not compliant with the [AMD specification](https://github.com/amdjs/amdjs-api/wiki/AMD#module-id-format-).

I filed https://github.com/Microsoft/TypeScript/issues/27908, which appears to be a duplicate of https://github.com/Microsoft/TypeScript/issues/16444.  See https://github.com/Microsoft/TypeScript/issues/16444 for more information.

Unfortunately, the following names are taken by similar projects:
* https://www.npmjs.com/package/aspectjs
* https://www.npmjs.com/package/aspect-js
* https://www.npmjs.com/package/aspect
* https://www.npmjs.com/package/aspects
* https://www.npmjs.com/package/aspectify

I noticed that `aspectj` is not currently taken, but you might run into trademark issues with [the AspectJ eclipse project](https://www.eclipse.org/aspectj/).

I also noticed that `aspectize` is not taken, so you may consider that, since it does have the word "aspect" in it, it would be easily found, and it sounds kind of javascript-y.

**Comment by matthewadams** [comment]
I agree on a non-scoped package.  Not sure why you're referring to packages that already exist.

IMHO, the name should maximize the probability of a hit when searched for, which is why I recommended `aspectize`, because it still has the word "aspect" in it.

How about `aspectomatic` or `aspect-o-matic`?  `aspecter`?  `slicer-dicer`?  [`slapchop`](https://www.slapchop.com/)?  `adviser` (sic)?

**Comment by mgechev** [comment]
>Not sure why you're referring to packages that already exist.

The packages crosscutjs and crosscut-js do not exist. We can have "aspect" as a keyword.

**Comment by matthewadams** [comment]
I'm partial to `aspectomatic`, 'cause it's so gosh darned funny.  If you want to go with a cross-cutting theme, you could go with a famous knife name, like `ginsu`, `orcrist` or anything from https://en.wikipedia.org/wiki/List_of_historical_swords.

**Comment by mgechev** [comment]
Works for me. Here's a [demo](https://github.com/mgechev/aspect.js-demo). Closing the issue, for now, feel free to reopen it and share a counterexample.

**Comment by eddiemonge** [comment]
Why is the issue closed? This doesn't seem like it was resolved.
